image_data_sar/review.py

```diff
@@ -0,0 +1,204 @@
+            obs['orig_text'] = chunk
+            catalogs.append(obs)
+    return catalogs
+
+
+def starcheck_for_obsid(obsid, sc):
+    for cat in sc:
+        if cat['obsid'] == int(obsid):
+            return cat
+
+
+def obs1_setup_ok(sc_cat):
+    imgsz = np.array([row['sz'] for row in sc_cat['catalog']])
+    return ((sc_cat['obs']['dither_y_amp'] == 16.0)
+            & (sc_cat['obs']['dither_z_amp'] == 16.0)
+            & (sc_cat['obs']['dither_y_period'] == 2000.0)
+            & (sc_cat['obs']['dither_z_period'] == 1414.8)
+            & (np.all(imgsz == '8x8')))
+
+
+def obs2_setup_ok(sc_cat):
+    imgsz = np.array([row['sz'] for row in sc_cat['catalog']])
+    return ((sc_cat['obs']['dither_y_amp'] == 8.0)
+            & (sc_cat['obs']['dither_z_amp'] == 8.0)
+            & (sc_cat['obs']['dither_y_period'] == 1000.0)
+            & (sc_cat['obs']['dither_z_period'] == 707.1)
+            & (np.all(imgsz == '6x6')))
+
+
+def obs3_setup_ok(sc_cat):
+    imgsz = np.array([row['sz'] for row in sc_cat['catalog']])
+    return ((sc_cat['obs']['dither_y_amp'] == 8.0)
+            & (sc_cat['obs']['dither_z_amp'] == 8.0)
+            & (sc_cat['obs']['dither_y_period'] == 1000.0)
+            & (sc_cat['obs']['dither_z_period'] == 707.1)
+            & (np.all(imgsz == '8x8')))
+
+
+def has_sc_critical(obs_text, ok6x6=False):
+    warnlines = [w for w in obs_text.split("\n")
+                 if re.compile(r"^\>\>\s+CRITICAL.*").match(w)]
+    if ok6x6:
+        warnlines = [w for w in warnlines if '6x6 Should be 8x8' not in w]
+    return len(warnlines) > 0
+
+
+def has_proseco_critical(aca):
+    acar = aca.get_review_table()
+    acar.run_aca_review()
+    return len(acar.messages == 'critical') > 0
+
+
+def get_durations(aca_arr):
+    durations = []
+    for aca, next_aca in zip(aca_arr[:-1], aca_arr[1:]):
+        if isinstance(next_aca, dict):
+            dwell_end = next_aca['sched_stop']
+        else:
+            man_dur = duration(aca.att, next_aca.att)
+            dwell_end = CxoTime(next_aca.date) - man_dur * u.s
+        durations.append(dwell_end - CxoTime(aca.date))
+    return durations
+
+
+def check_duration(obsid, obsids, durations):
+    idx = np.flatnonzero(obsids == obsid)[0]
+    return durations[idx] > 4000 * u.s
+
+
+def same_guide_stars(obs1, obs2, obs3, acas):
+    gs1 = list(acas[obs1].guides['id'])
+    gs2 = list(acas[obs2].guides['id'])
+    gs3 = list(acas[obs3].guides['id'])
+    return set(gs1) == set(gs2) == set(gs3)
+
+
+def same_attitude(obs1, obs2, obs3, acas):
+    att1 = acas[obs1].call_args['att']
+    att2 = acas[obs2].call_args['att']
+    att3 = acas[obs3].call_args['att']
+    dq2 = att1.dq(att2)
+    dq3 = att1.dq(att3)
+    return ((np.abs(dq2.pitch * 3600) < 1)
+            and (np.abs(dq2.yaw * 3600) < 1)
+            and (np.abs(dq3.pitch * 3600) < 1)
+            and (np.abs(dq3.yaw * 3600) < 1)
+            and np.allclose([att2.roll, att3.roll], att1.roll, atol=0.05, rtol=0))
```

Comment:
With regard to attitude, practically, I don't recall from ORViewer if there is a limit on how identical the attitudes can be.  If they can be identical, maybe we should just check for that.  What are your thoughts @taldcroft ?